DZ_1

Belchitta opened the pull request:
Lesson-1

Belchitta committed:
DZ_1

Lesson-1

DmitryChitalov commented on Lesson-1/task_1.py:
```diff
@@ -0,0 +1,63 @@
+    Сложность: O(n).
+    """
+    lst_to_set = set(lst_obj)  # O(n) - линейная сложность
+    return lst_to_set  # O(1) - константная сложность
```
выполнено

DmitryChitalov commented on Lesson-1/task_1.py:
```diff
@@ -0,0 +1,63 @@
+        if lst_obj[j] in lst_obj[j+1:]:    # O(n) линейная сложность
+            return False                   # O(1) константная сложность
+    return True                            # O(1) константная сложность
+
```
выполнено

DmitryChitalov commented on Lesson-1/task_1.py:
```diff
@@ -0,0 +1,63 @@
+        if lst_copy[i] == lst_copy[i+1]:     # O(1) константная сложность
+            return False                     # O(1) константная сложность
+    return True                              # O(1) константная сложность
+
```
выполнено

DmitryChitalov commented on Lesson-1/task_2.py:
```diff
@@ -0,0 +1,41 @@
+lst_ = [randint(0, 100) for i in range(20)]
+print(lst_)
+print(list_min(lst_))
+print(list_min_2(lst_))
```
выполнено

DmitryChitalov commented on Lesson-1/task_3.py:
```diff
@@ -0,0 +1,53 @@
+def sorted_2(capitals):
+    max_income = {}                                                  # O(1)
+    top_capitals = dict(capitals)                                    # O(1)
+    for i in range(3):                                               # O(n)
```
for i in range(3):                                               # O(n)
здесь константная

DmitryChitalov replied:
max( - какая сложность у max?

DmitryChitalov commented on Lesson-1/task_4.py:
```diff
@@ -0,0 +1,66 @@
+print(authorization_check1(users, 'betha', 'b123'))
+print(authorization_check1(users, 'gamma', 'g000'))
+
+# Вывод: решение с константной сложностью O(1) лучше второго с O(n)
```
выполнено

DmitryChitalov commented on Lesson-1/task_5.py:
```diff
@@ -0,0 +1,48 @@
+    print(SC_OBJ.is_empty())
+    SC_OBJ.push_in(4.4)
+    print(SC_OBJ.stack_size())
+    print(SC_OBJ.elems)
```
выполнено

DmitryChitalov commented on Lesson-1/task_6.py:
```diff
@@ -0,0 +1,55 @@
+После реализации структуры, проверьте ее работу на различных сценариях
+Примечание: ПРОШУ ВАС ВНИМАТЕЛЬНО ЧИТАТЬ ЗАДАНИЕ!
+"""
+class taskboard:
```
taskboard
нарушение пеп-8 в именах классов

DmitryChitalov commented on Lesson-1/task_7.py:
```diff
@@ -0,0 +1,58 @@
+    return still_equal
+
+
+print(pal_checker("молоко делили ледоколом"))
```
выполнено

Belchitta committed:
DZ_1_Доработаны task 3, 6

task_3: с циклом поняла, ещё раз посмотрела таблицы, вижу, что max - O(n)
task_6: Исправилась, спасибо, что не снизили оценку до минимальной. Ещё раз повторю основы про классы.
